crates/resolver/src/name_server/connection_provider.rs

```diff
@@ -57,27 +70,45 @@ pub trait ConnectionProvider: 'static + Clone + Send + Sync + Unpin {
         -> Self::FutureConn;
 }
 
-fn spawn_bg<F: Future<Output = Result<(), ProtoError>> + Send + 'static>(
-    spawner: &Handle,
-    background: F,
-) {
-    // TODO: maybe do something with this in the future?
-    let _join = spawner.spawn(background);
+/// A type defines runtime.
+pub trait RuntimeProvider: Clone + 'static {
+    /// Handle to the executor;
+    type Handle: Clone + Send + Spawn + Sync + Unpin;
+
+    /// TcpStream
+    type Tcp: AsyncRead + AsyncWrite + Connect + Send + Unpin;
+
+    /// Timer
+    type Timer: Send + Time + Unpin;
+
+    /// UdpSocket
+    type Udp: Send + UdpSocket;
+}
+
+/// A type defines the Handle which can spawn future.
+pub trait Spawn {
+    fn spawn_bg(
+        &mut self,
+        future: Pin<Box<dyn Future<Output = Result<(), ProtoError>> + Send + 'static>>,
```

Comment:
does this have to be a `Pin<Box>>`? or could be just make spawn_bg generic with `spawn_bg<F>(&mut self, future: F) where F: Future<Output = Result<(), ProtoError>> + Send + 'static>`

Not a huge deal, but I think it would be easier to use if it was a generic function.

Reply:
Updated.